### PR TITLE
Fixed hologram scaling for ragdolls, models with bodygroups and some other models

### PR DESCRIPTION
**Pull request by AlexanderArvidsson**

### CODE REVIEW

File: lua/entities/gmod_starfall_hologram/cl_init.lua [hunk 1]
```diff
@@ -121,9 +121,19 @@ end )
...
+	if boneCount > 1 then
+		for i = 0, boneCount do
+			self:ManipulateBoneScale( i, scale )
+		end
+	elseif self.EnableMatrix then
+		local m = Matrix()
+		m:Scale( scale )
+		self:EnableMatrix( "RenderMultiply", m )
+	else
+		self.SetModelScale( ( scale.x + scale.y + scale.z ) / 3, 0 )
```
Comment (ColonelThirtyTwo) [inline lua/entities/gmod_starfall_hologram/cl_init.lua]: Shouldn't this be `self:SetModelScale()`? Otherwise, how does it get the entity?

**Comment by AlexanderArvidsson** [comment]
Oh crap, that's a typo, i'll change that when I get home in some hours

Edit: Fixed it with teamviewer

File: lua/entities/gmod_starfall_hologram/cl_init.lua [hunk 2]
```diff
@@ -121,9 +121,19 @@ end )
 -- @param scale Vector scale
 function ENT:SetScale ( scale )
 	self.scale = scale
-	local m = Matrix()
-	m:Scale( scale )
-	self:EnableMatrix( "RenderMultiply", m )
+	
+	local boneCount = self:GetBoneCount() or -1
+	if boneCount > 1 then
+		for i = 0, boneCount do
```
Comment (Xandaros) [inline lua/entities/gmod_starfall_hologram/cl_init.lua:127]: Aren't you modifying one extra bone?
You count from 0 to boneCount, which is boneCount+1...

Also, it still looks ugly with ragdolls, but I'm not sure what to do about it. entity:SetModelScale does it properly, but it only works with scalars :/
(And E2 has the same problem)

**Comment by AlexanderArvidsson** [comment]
While i'm pretty sure changing the 0 to a 1 would be correct, expression 2 scales bones from 0 to bone count. Either wiremod did it wrong, or there's a reason behind it. Either way, this works, and wiremod does the same.
I haven't tested it with 1, but it probably won't be different.

**Comment by awilliamson** [comment]
So you copied it straight from E2, did no testing, did no thinking? I at least expected some preliminary testing to be done, it makes our lives a lot easier when coming to accept them.

Not everything in E2/Wiremod is 100% working, that's why issues exist, at least check that something works prior to copying.

**Comment by AlexanderArvidsson** [comment]
I never said I didn't do any testing. I tested it several times on several models. I didn't copy-paste it if that's what you think. I only checked what had to be done and then wrote it similarly to get things done quickly and not write my own which could be buggy. There's a common quote that goes something like this:
"Good programmers write good code; Great programmers copy great code."
Totally not saying wiremod's code is great, but you get the point.

Since I don't really know how bones work in garry's mod, I included index 0 just to be sure if some models had their bones on 0.
I'm only trying to contribute to the project.
Anyway, sorry if this has upset you, I do not want this to turn into a big discussion :P

**Comment by Xandaros** [comment]
Actually, bones are 0-indexed, so the first bone is 0. However, since there are boneCount bones and 0 is a valid index, the last index is boneCount-1.
So it needs to be `for i = 0, boneCount - 1 do`

However, this does not actually look very good... at least on Kleiner, are there any ragdolls where this actually looks ok?

**Pull request closed without merging**